collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/EntityFormTest.kt

```diff
@@ -90,54 +96,39 @@ class EntityFormTest {
     }
 
     @Test
-    fun fillingEntityFollowUpForm_whenOnlineDuplicateHasHigherVersion_showsOnlineVersion() {
-        testDependencies.server.addForm("one-question-entity-update.xml", listOf("people.csv"))
-
-        val mainMenuPage = rule.withMatchExactlyProject(testDependencies.server.url)
+    fun entityListsAreConsistentBetweenFollowUpForms() {
+        testDependencies.server.apply {
+            addForm(
+                "one-question-entity-update.xml",
+                listOf("people.csv")
+            )
+
+            addForm(
+                "one-question-entity-follow-up.xml",
+                mapOf("people.csv" to "updated-people.csv")
+            )
+        }
+
+        rule.withProject(testDependencies.server)
             .enableLocalEntitiesInForms()
+            .addEntityListInBrowser("people")
 
+            .clickGetBlankForm()
+            .clickClearAll()
+            .clickForm("one-question-entity-update.xml")
+            .clickGetSelected()
+            .clickOK(MainMenuPage())
             .startBlankForm("One Question Entity Update")
-            .assertQuestion("Select person")
-            .clickOnText("Roman Roy")
-            .swipeToNextQuestion("Name")
-            .answerQuestion("Name", "Romulus Roy")
-            .swipeToEndScreen()
-            .clickFinalize()
-
-        testDependencies.server.updateMediaFile(
-            "one-question-entity-update.xml",
-            "people.csv",
-            "updated-people.csv"
-        )
+            .assertText("Roman Roy")
+            .pressBackAndDiscardForm()
 
-        mainMenuPage.clickFillBlankForm()
-            .clickRefresh()
-            .clickOnForm("One Question Entity Update")
+            .clickGetBlankForm()
+            .clickClearAll()
```

Comment:
I want to specifically download 'one-question-entity-follow-up.xml` so that I check that downloading `updated-people.csv` updates the list. I guess a potentially simpler way to test this is to just download both forms at once and check they have the same list, but I'd prefer to test the extreme case here: forms being downloaded individually with different entity lists. Does that make sense?

Reply:
Yes, it makes sense but my question was about something else. Once you download the first form and open the list of forms to download again the one you already have should be unselected by default. It seems like both are selected as if the first one was removed or not downloaded at all yet. You shouldn't need to click the clear button and then select that form because the form you want to download should be the only one selected.

Reply:
Ah got you! I think I'd still prefer the "clear all" to make it explicit I'm selecting that form.

Reply:
> Ah got you! 

No, I think you don't. Please run that test `entityListSecondaryInstancesAreConsistentBetweenFollowUpForms` and add a breakpoint in line `126` before you call `clearAll()`. I was asking if it's a bug that both forms are selected by default after opening the list of forms to download when one of them is already downloaded (line 119).

Reply:
Sorry! That'll be the brain fog last week. I see what you mean: it's weird that both forms are selected. I'm guessing this is a problem with `StubOpenRosaServer` rather an actual bug, but I'll look into it.

Reply:
Yeah it's because `StubOpenRosaServer` uses the file path as the form ID for the form list, whereas Collect doesn't trust the server and parses it out of the form itself. That means we end up in a scenario where the form gets downloaded and saved with the ID in the XML, but the server is still using the "fake form ID". I've fixed it so that `StubOpenRosaServer` will parse the form ID out of the XForm if it's not given an explicit form ID.